### PR TITLE
Add section for new Foreign Key Constraints runtime setting

### PR DESCRIPTION
**Pull request by passalaqua**
This setting will be added in Mendix 10.10.

### CODE REVIEW

File: content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md [hunk 1]
```diff
@@ -215,6 +215,16 @@ To force a query to the runtime, use microflows. For example, create a microflow
 
 Default: *Yes*
 
+### 3.15 Foreign Key Constraints {#database-fkc}
```
Comment (MarkvanMents) [inline content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md:218]: Should we refer to this in https://docs.mendix.com/refguide/data-storage/#fkc?

File: content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md [hunk 2]
```diff
@@ -215,6 +215,16 @@ To force a query to the runtime, use microflows. For example, create a microflow
...
+
+If this option is enabled, database [foreign key constraints](/refguide/data-storage/#fkc) will be used. An attempt to commit a dangling reference will throw a runtime exception.
+
+{{% alert color="info" %}}
+This option was added in Mendix version 10.10. From Mendix versions 10.6 to 10.9, foreign key constraints are always used. Mendix versions below 10.6 do not use foreign key constraints.
+{{% /alert %}}
+
+Default: *Yes* (for projects created with Mendix versions 10.6.0 and above)
```
Comment (MarkvanMents) [inline content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md]: What happens when upgrading from, say, 9.24 or 10.4 to 10.10? Is there a default set - what will it be?

File: content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md [hunk 3]
```diff
@@ -215,6 +215,16 @@ To force a query to the runtime, use microflows. For example, create a microflow
 
 Default: *Yes*
 
+### 3.15 Foreign Key Constraints {#database-fkc}
+
+If this option is enabled, database [foreign key constraints](/refguide/data-storage/#fkc) will be used. An attempt to commit a dangling reference will throw a runtime exception.
+
+{{% alert color="info" %}}
+This option was added in Mendix version 10.10. From Mendix versions 10.6 to 10.9, foreign key constraints are always used. Mendix versions below 10.6 do not use foreign key constraints.
```
Comment (MarkvanMents) [inline content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md]: Are my additions here accurate?

**Review by MarkvanMents — CHANGES_REQUESTED** [review]
Thanks for the PR, Augusto.
I've made some additions and asked a couple of questions below. Can you review them?
Thanks.

File: content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md (hunk 3 above)
Comment (passalaqua) [reply content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md]: No. It really depends on which version the project was originally created. See the reply regarding Defaults.

File: content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md (hunk 2 above)
Comment (passalaqua) [reply content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md]: If it was not created in 10.6 or above, then it is not enabled. That means upgrades should not affect the previous value.

Do you have a suggestion on how to write this without making it verbose and confusing?

File: content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md (hunk 3 above)
Comment (MarkvanMents) [reply content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md]: OK, so we should put it back to this?
```suggestion
This option was added in Mendix version 10.10.
```

File: content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md (hunk 2 above)
Comment (MarkvanMents) [reply content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md]: Can we just say the following?
```suggestion
Default: *depends on the version of Mendix used to create the app:*

* *Yes* for apps created with Mendix versions 10.6.0 and above
* *No* for apps created with Mendix versions below 10.6
```
I assume that once they get to 10.10 they will be able to turn it on/off independently of which version was used to create the app.

**Review by MarkvanMents — CHANGES_REQUESTED** [review]
Thanks for the answers.
I've made a couple of suggestions for changes. Can you look at them and commit them if they look OK, or let me know if they still aren't clear/correct.
Mark.

File: content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md (hunk 3 above)
Comment (passalaqua) [reply content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md]: I think putting it back is better, indeed.

File: content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md (hunk 2 above)
Comment (passalaqua) [reply content/en/docs/refguide/modeling/app-explorer/app/app-settings/_index.md]: Yes, turning on/off is precisely the point of exposing this setting.

**Comment by dahfjkg** [comment]
Passed technical review

**Review by MarkvanMents — APPROVED** [review]
Ready for publication with Mendix 10.10
Thanks for the PR, Augusto.

**Pull request merged**